From e58947650da648961d5e9c899ad2b95d2e8b20dd Mon Sep 17 00:00:00 2001
From: Andrew Boie <andrew.p.boie@intel.com>
Date: Tue, 25 Sep 2012 14:53:46 -0700
Subject: x86: add HSW/VLV build tuning directives

This is just the Ivy Bridge and x86-atom configs for now,
needs to be tuned later once we have compiler support.

Change-Id: Iacaffccff728ced20fd08c212f2ad3209d31fdb6
Signed-off-by: Andrew Boie <andrew.p.boie@intel.com>
---
 core/combo/arch/x86/haswell.mk    |   22 ++++++++++++++++++++++
 core/combo/arch/x86/valleyview.mk |   20 ++++++++++++++++++++
 2 files changed, 42 insertions(+), 0 deletions(-)
 create mode 100644 core/combo/arch/x86/haswell.mk
 create mode 100644 core/combo/arch/x86/valleyview.mk

diff --git a/core/combo/arch/x86/haswell.mk b/core/combo/arch/x86/haswell.mk
new file mode 100644
index 0000000..45c8e29
--- /dev/null
+++ b/core/combo/arch/x86/haswell.mk
@@ -0,0 +1,22 @@
+# Configuration for Linux on x86.
+# Generating binaries for Haswell processors.
+# FIXME: This is just ivy bridge; update for Haswell
+# capabilities once we have compiler support
+
+ARCH_X86_HAVE_MMX    := true
+ARCH_X86_HAVE_SSE    := true
+ARCH_X86_HAVE_SSE2   := true
+ARCH_X86_HAVE_SSE3   := true
+ARCH_X86_HAVE_SSSE3  := true
+ARCH_X86_HAVE_SSE4   := true
+ARCH_X86_HAVE_SSE4_1 := true
+ARCH_X86_HAVE_SSE4_2 := true
+ARCH_X86_HAVE_AES_NI := true
+ARCH_X86_HAVE_AVX    := true
+
+# CFLAGS for this arch
+arch_variant_cflags := \
+	-march=corei7-avx \
+	-mstackrealign \
+	-mfpmath=sse \
+
diff --git a/core/combo/arch/x86/valleyview.mk b/core/combo/arch/x86/valleyview.mk
new file mode 100644
index 0000000..1ea4eea
--- /dev/null
+++ b/core/combo/arch/x86/valleyview.mk
@@ -0,0 +1,20 @@
+# This file contains feature macro definitions specific to the
+# 'x86-atom' arch variant. This is an extension of the 'x86' base variant
+# that adds Atom-specific features.
+#
+# See build/core/combo/arch/x86/x86.mk for differences.
+#
+ARCH_X86_HAVE_MMX   := true
+ARCH_X86_HAVE_SSE   := true
+ARCH_X86_HAVE_SSE2  := true
+ARCH_X86_HAVE_SSE3  := true
+ARCH_X86_HAVE_SSSE3 := true
+ARCH_X86_HAVE_MOVBE := true
+ARCH_X86_HAVE_POPCNT := false   # popcnt is not supported by current Atom CPUs
+
+# CFLAGS for this arch
+arch_variant_cflags := \
+	-march=atom \
+	-mstackrealign \
+	-mfpmath=sse \
+
-- 
1.7.4.1

